src/app/users/user-list/user-list.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import { UserService } from '../user.service';
+import { AsyncPipe } from '@angular/common';
+import { UserItemComponent } from "../user-item/user-item.component";
+import { ActivatedRoute } from '@angular/router';
+import { AuthService } from '../../auth/auth.service';
+
+@Component({
+  selector: 'app-user-list',
+  standalone: true,
+  imports: [
+    AsyncPipe,
+    UserItemComponent
+],
+  templateUrl: './user-list.component.html',
+  styleUrl: './user-list.component.scss'
+})
+export class UserListComponent implements OnInit {
+
+  userList = this.userService.usersSubject;
+  userList$ = this.userService.usersSubject.asObservable();
+
+  constructor(private userService: UserService, private route: ActivatedRoute, private authService: AuthService) {}
+
+  ngOnInit(): void {
+
+    // Check if logged in user is admin, or if not then compare ids
+    let loggedInUser = this.authService.currentUserSubject.getValue();
+    if(!loggedInUser || loggedInUser.role.value != "ADMIN") throw Error("Only users with administrator priviledges have access.");
```

Comment:
you could add this logic in the page guard as it's related to permissions of accessing the page

Reply:
I decided on my implementation because the route /users/:userId should be accessible for the user itself so that they can view their own user details page. But putting a guard that checks if the user is admin on the route /users, makes the route /users/:userId not accessible even if the user is checking their own details. Both routes start with /users and admin guard gets called by default.

Reply:
you could check in the guard if the user is the same one we are accessing or if the role is admin then we let the user access the page, what do you think?